### PR TITLE
Donut-Vend add

### PR DESCRIPTION
**Pull request by Deahaka**
## About the PR 
Add donut-vend for heavy roleplay officers. There are not many regular donuts in the build that would be worth inserting into the machine, so there are not enough of them now.
by [Theta Station](https://github.com/ThetaStation/ThetaStation).
**Screenshots**

**Changelog**

:cl:
- add: Added donut-vend.

### CODE REVIEW

**Comment by Deahaka** [comment]
image

**Comment by moonheart08** [comment]
> 
> 
> image

Excuse me?

**Comment by Deahaka** [comment]
> Excuse me? Is there an issue here.

no, i'm just wondering why it works like this?

**Comment by moonheart08** [comment]
> > Excuse me? Is there an issue here.
> 
> no, i'm just wondering why it works like this?

Remie is a codeowner on something you added, so she gets added to the reviewer list.

File: Resources/Locale/en-US/advertisements/vending/donut.ftl [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿advertisement-donut-1 = Homer Simpson approves!
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/advertisements/vending/donut.ftl:1]: this probably doesn't matter that much but generally we want to stay copyright safe and avoid references to intellectual properties.

File: Resources/Prototypes/Entities/Structures/Machines/vending_machines.yml [hunk 2]
```diff
@@ -887,6 +887,36 @@
...
+      shader: unshaded
+    - state: panel
+      map: ["enum.WiresVisualLayers.MaintenancePanel"]
+  - type: Appearance
+    visuals:
+    - type: VendingMachineVisualizer
+      normalUnshaded: true
+      broken: true
+    - type: WiresVisualizer
+
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Structures/Machines/vending_machines.yml:919]: maybe give it a point light like the other vendors?

**Comment by Deahaka** [comment]
**Done**

File: Resources/Prototypes/Catalog/VendingMachines/Advertisements/donut.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- type: advertisementsPack
+  id: DonutAds
+  advertisements:
+    - advertisement-donut-1
+    - advertisement-donut-2
+    - advertisement-donut-3
+    - advertisement-donut-4
```
Comment (mirrorcult) [inline Resources/Prototypes/Catalog/VendingMachines/Advertisements/donut.yml:7]:
```suggestion
```

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/donut.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+- type: vendingMachineInventory
+  id: DonutInventory
+  name: Donut
+  spriteName: donut
+  startingInventory:
+    FoodDonutChocolate: 5
+    FoodDonutApple: 5
+    FoodDonutPink: 5
+    FoodDonutBungo: 5
```
Comment (mirrorcult) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/donut.yml:9]: should look something like this for inventory imo, chocolate are the most common

```suggestion
    FoodDonutChocolate: 5
    FoodDonutApple: 3
    FoodDonutPink: 3
    FoodDonutBungo: 3
```

**Review by mirrorcult — CHANGES_REQUESTED** [review]
looks great outside of that, will probably have to be mapped sparsely though

**Comment by Deahaka** [comment]
Fixed as requested in review

**Comment by Partmedia** [comment]
Presently, you can hack security vending machines to dispense donuts. If donut dispensers become more commonplace, should donuts be removed from the hacked security vendor list?

**Comment by Deahaka** [comment]
Maybe, I just thought that it would be necessary to give donuts with cyanide for a hacked donut-vend

**Comment by Deahaka** [comment]
> Presently, you can hack security vending machines to dispense donuts. If donut dispensers become more commonplace, should donuts be removed from the hacked security vendor list?

I looked, it contains a reference and for some reason boxes of donuts. I can remove the boxes, but I think the reference should be left there.

**Pull request merged**